src/Unirest/Request.php

```diff
@@ -22,6 +22,16 @@ class Request
     private static $verifyPeer = true;
     private static $verifyHost = true;
 
+    /**
+     * Should enable or disable retries for current request?
+     * True:  Retry current request ignoring httpMethods whitelist.
+     * False: Do not retry current request.
+     * Null:  Use global httpMethods whitelist to determine if it needs retrying.
+     *
+     * @var bool|null
+     */
+    private static $retryCurrentRequest = null;
```

Comment:
I suggest we go with a slightly better name of `overrideRetryForNextRequest`.

Also, let's go from using a nullable boolean type to using an enum so that the values are more self-explanatory.

For example, checkout these calls:
1. `Unirest::overrideRetryForNextRequest(OverrideRetry::ENABLE_RETRY)`
2. `Unirest::overrideRetryForNextRequest(OverrideRetry::DISABLE_RETRY)`
3. `Unirest::overrideRetryForNextRequest(OverrideRetry::USE_GLOBAL_SETTINGS)`

Reply:
Changes done!